src/cascadia/TerminalSettingsModel/GlobalAppSettings.cpp

```diff
@@ -134,6 +138,11 @@ void GlobalAppSettings::LayerJson(const Json::Value& json, const OriginTag origi
     // "useTabSwitcher", but prefer "tabSwitcherMode"
     JsonUtils::GetValueForKey(json, LegacyUseTabSwitcherModeKey, _TabSwitcherMode);
 
+    JsonUtils::GetValueForKey(json, LegacyInputServiceWarningKey, _InputServiceWarning);
+    JsonUtils::GetValueForKey(json, LegacyWarnAboutLargePasteKey, _WarnAboutLargePaste);
+    JsonUtils::GetValueForKey(json, LegacyWarnAboutMultiLinePasteKey, _WarnAboutMultiLinePaste);
+    JsonUtils::GetValueForKey(json, LegacyConfirmCloseAllTabsKey, _ConfirmCloseAllTabs);

```

Comment:
When we encounter such a legacy key, I think we should trigger a settings save so that we can remove these keys in the near-ish term.